macros/edr/dbt_artifacts/upload_artifacts_to_table.sql

```diff
@@ -1,18 +1,39 @@
-{% macro upload_artifacts_to_table(table_relation, artifacts, flatten_artifact_callback, truncate_if_on_run_end, should_commit=False) %}
+{% macro upload_artifacts_to_table(table_relation, artifacts, flatten_artifact_callback, append=False, should_commit=False) %}
     {% set flatten_artifact_dicts = [] %}
     {% for artifact in artifacts %}
         {% set flatten_artifact_dict = flatten_artifact_callback(artifact) %}
-        {% if flatten_artifact_dict is not none %}
+        {% if flatten_artifact_dict is not none and elementary.dict_has_nonempty_values(flatten_artifact_dict) %}
```

Comment:
What is this check needed for?
When would `dict_has_nonempty_values()` be `true`?

Reply:
I'm guessing you mean when it would be `false`?
I'm not sure to be honest
I added it to replace `remove_empty_rows` which did the same logic in SQL after the table was created.

Reply:
Maybe let's remove this if we don't understand it?
I don't like how we have "paranoid" code that we can't give a concrete reason for its existence.

Reply:
👍 
I'll change